src/core_codemods/https_connection.py

```diff
@@ -1,135 +1,82 @@
-from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
-from libcst.metadata import (
-    PositionProvider,
-)
-from codemodder.codemods.base_codemod import (
-    BaseCodemod,
-    CodemodMetadata,
-    ReviewGuidance,
-)
-from codemodder.change import Change
-from codemodder.codemods.utils_mixin import NameResolutionMixin
-from codemodder.file_context import FileContext
+
 import libcst as cst
-from libcst.codemod import (
-    Codemod,
-    CodemodContext,
-    VisitorBasedCodemodCommand,
-)
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.metadata import PositionProvider
 
+from codemodder.codemods.base_codemod import ReviewGuidance
+from codemodder.codemods.api import BaseCodemod
+from codemodder.codemods.imported_call_modifier import ImportedCallModifier
 
-class HTTPSConnection(BaseCodemod, Codemod):
-    METADATA = CodemodMetadata(
-        DESCRIPTION="Enforce HTTPS connection for `urllib3`.",
-        NAME="https-connection",
-        REVIEW_GUIDANCE=ReviewGuidance.MERGE_WITHOUT_REVIEW,
-        REFERENCES=[
-            {
-                "url": "https://owasp.org/www-community/vulnerabilities/Insecure_Transport",
-                "description": "",
-            },
-            {
-                "url": "https://urllib3.readthedocs.io/en/stable/reference/urllib3.connectionpool.html#urllib3.HTTPConnectionPool",
-                "description": "",
-            },
-        ],
-    )
-    CHANGE_DESCRIPTION = METADATA.DESCRIPTION
-    SUMMARY = (
-        "Changes HTTPConnectionPool to HTTPSConnectionPool to Enforce Secure Connection"
-    )
+
+class HTTPSConnectionModifier(ImportedCallModifier):
+    def updated_args(self, original_args):
+        """
+        Last argument _proxy_config does not match new method
+
+        We convert it to keyword
+        """
+        new_args = list(original_args)
+        if self.count_positional_args(new_args) == 10:
+            new_args[9] = new_args[9].with_changes(
+                keyword=cst.parse_expression("_proxy_config")
+            )
+        return new_args
+
+    def update_attribute(self, true_name, original_node, updated_node, new_args):
+        del true_name, original_node
+        return updated_node.with_changes(
+            args=new_args,
+            func=updated_node.func.with_changes(
+                attr=cst.parse_expression("HTTPSConnectionPool")
+            ),
+        )
+
+    def update_simple_name(self, true_name, original_node, updated_node, new_args):
+        del true_name
+        AddImportsVisitor.add_needed_import(self.context, "urllib3")
+        RemoveImportsVisitor.remove_unused_import_by_node(self.context, original_node)
+        return updated_node.with_changes(
+            args=new_args,
+            func=cst.parse_expression("urllib3.HTTPSConnectionPool"),
+        )
+
+    def count_positional_args(self, arglist: Sequence[cst.Arg]) -> int:
+        for i, arg in enumerate(arglist):
```

Comment:
nitpick and personal preference to not use `i` and instead use `idx` (easier to grep)